src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupBackupProvider.cs

```diff
@@ -57,7 +61,328 @@ public class AzureFilesPsBackupProvider : IPsBackupProvider
+                    protectableObjectResource = protectableItem;
+                    break;
+                }
+            }
+            return protectableObjectResource;
+        }
+
+        private void TriggerInquiry(string vaultName, string resourceGroupName,
+            string storageContainer)
+        {
+            ODataQuery<BMSContainersInquiryQueryObject> queryParam = new ODataQuery<BMSContainersInquiryQueryObject>(
+                q => q.WorkloadType
+                     == ServiceClientModel.WorkloadType.AzureFileShare);
+            string errorMessage = string.Empty;
+            var inquiryResponse = ServiceClientAdapter.InquireContainers(
+               storageContainer,
+               queryParam,
+               vaultName,
+               resourceGroupName);
+
+            var operationStatus = TrackingHelpers.GetOperationResult(
+                inquiryResponse,
+                operationId =>
+                    ServiceClientAdapter.GetRefreshContainerOperationResult(
+                        operationId,
+                        vaultName: vaultName,
+                        resourceGroupName: resourceGroupName));
+
+            //Now wait for the operation to Complete
+            if (inquiryResponse.Response.StatusCode
+                    != SystemNet.HttpStatusCode.NoContent)
+            {
+                errorMessage = string.Format(Resources.DiscoveryFailureErrorCode,
+                    inquiryResponse.Response.StatusCode);
+                Logger.Instance.WriteDebug(errorMessage);
+            }
+        }
+
+        private List<ContainerBase> GetRegisteredStorageAccount(string vaultName = null, string resourceGroupName = null)
+        {
+            ODataQuery<BMSContainerQueryObject> queryParams = null;
+            queryParams = new ODataQuery<BMSContainerQueryObject>(
+                q => q.BackupManagementType == ServiceClientModel.BackupManagementType.AzureStorage);
+
+            var listResponse = ServiceClientAdapter.ListContainers(
+                queryParams,
+                vaultName: vaultName,
+                resourceGroupName: resourceGroupName);
+
+            List<ContainerBase> containerModels = ConversionHelpers.GetContainerModelList(listResponse);
+
+            return containerModels;
+        }
+
+        private List<ProtectableContainerResource> GetUnRegisteredStorageAccount(string vaultName = null,
+            string resourceGroupName = null)
+        {
+            ODataQuery<BMSContainerQueryObject> queryParams = null;
+            queryParams = new ODataQuery<BMSContainerQueryObject>(
+                q => q.BackupManagementType == ServiceClientModel.BackupManagementType.AzureStorage);
+
+            var listResponse = ServiceClientAdapter.ListUnregisteredContainers(
+                queryParams,
+                vaultName: vaultName,
+                resourceGroupName: resourceGroupName);
+            List<ProtectableContainerResource> containerModels =
```

Comment:
Why are we converting these models to PS models? Can't we work without converting them?